### PR TITLE
Add a "boot" phase

### PR DESCRIPTION
**Pull request by DanielCoulbourne**
Adds a mount phase which happens before auth and validate.

This phase can be used to set up event props, etc.

Useful if you want to register custom attributes that modify properties like so:

```php
class SomeEvent extends Event
{
    #[NowIfNull] // We can register a listener on the mount hook that sets this to now() if it is not passed.
    public ?Carbon $when_did_the_thing_happen = null;
}
```

### CODE REVIEW

File: src/Lifecycle/Dispatcher.php [hunk 1]
```diff
@@ -85,6 +96,19 @@ public function replay(Event $event): void
         }
     }
 
+    /** @return Collection<int, Hook> */
+    protected function getMountHooks(Event $event): Collection
+    {
+        $hooks = $this->hooksFor($event, Phase::Mount);
+
+        if (method_exists($event, 'mount')) {
+            dump($event);
```
Comment (inxilpro) [inline src/Lifecycle/Dispatcher.php:105]: Whoops :P

```suggestion
```

File: examples/Bank/src/Events/AccountOpened.php [hunk 2]
```diff
@@ -39,4 +40,12 @@ public function handle()
...
         Verbs::unlessReplaying(fn () => Mail::send(new WelcomeEmail($this->user_id)));
     }
+
+    public static function migrate()
+    {
+        return [
+            1 => fn (Collection $v0) => $v0->except('user_id'),
+            2 => fn (Collection $v1) => $v1->merge(['email' => 'default@email.gov']),
+        ];
+    }
```
Comment (inxilpro) [inline examples/Bank/src/Events/AccountOpened.php:50]: Is this supposed to be here?

**Review by inxilpro — COMMENTED** [review]
Can we call it `boot` or `construct` instead of `mount`?

**Comment by joshhanley** [comment]
Would it also run when an event is replayed? If so, then I think `boot()` would be better

**Comment by inxilpro** [comment]
Reviewed with Daniel. We need to address the differences between this and #145 but otherwise looks good.

**Pull request merged**